pkg/manifest/jsonmanifest/jsonmanifest_test.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package jsonmanifest_test
+
+import (
+	"net/http"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/manifest"
+	"github.com/ethersphere/bee/pkg/manifest/jsonmanifest"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/swarm/test"
+)
+
+var testCases = []testCase{
+	{
+		name:    "empty-manifest",
+		entries: nil,
+	},
+	{
+		name: "one-entry",
+		entries: []e{
+			{
+				reference: test.RandomAddress(),
+				name:      "entry-1",
+				headers:   http.Header{},
+				path:      "",
+			},
+		},
+	},
+	{
+		name: "two-entries",
+		entries: []e{
+			{
+				reference: test.RandomAddress(),
+				name:      "entry-1.txt",
+				headers:   http.Header{"Content-Type": {"text/plain; charset=utf-8"}},
+				path:      "",
+			},
+			{
+				reference: test.RandomAddress(),
+				name:      "entry-2.png",
+				headers:   http.Header{"Content-Type": {"image/png"}},
+				path:      "",
+			},
+		},
+	},
+	{
+		name: "nested-entries",
```

Comment:
what is nested here?

Reply:
the entries here represent the following structure:
```
nested-entries
├── readme.md
├── text/
│   └──robots.txt
└── img/
     ├── 1.png
     └── 2.png
```

Reply:
oh wow this wasn't so clear at first glance. i wonder what `path` even gives as an added value here. why not have just one `path` field that stores the full path string and just provide `Name()` on the fly by returning whatever is after the last slash?

Reply:
nice catch.

i've simplified the test case entry structure so that only `path` is used (which is a full path) and then the entry name is extracted from there.